prometheus/src/main/java/org/opensearch/sql/prometheus/storage/PrometheusStorageFactory.java

```diff
@@ -0,0 +1,93 @@
+/*
+ *
+ *  * Copyright OpenSearch Contributors
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.opensearch.sql.prometheus.storage;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import okhttp3.OkHttpClient;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.sql.catalog.model.ConnectorType;
+import org.opensearch.sql.catalog.model.auth.AuthenticationType;
+import org.opensearch.sql.prometheus.authinterceptors.AwsSigningInterceptor;
+import org.opensearch.sql.prometheus.authinterceptors.BasicAuthenticationInterceptor;
+import org.opensearch.sql.prometheus.client.PrometheusClient;
+import org.opensearch.sql.prometheus.client.PrometheusClientImpl;
+import org.opensearch.sql.storage.StorageEngine;
+import org.opensearch.sql.storage.StorageEngineFactory;
+
+public class PrometheusStorageFactory implements StorageEngineFactory {
+
+  private static final Logger LOG = LogManager.getLogger();
+  public static final String URI = "prometheus.uri";
+  public static final String AUTH_TYPE = "prometheus.auth.type";
+  public static final String USERNAME = "prometheus.auth.username";
+  public static final String PASSWORD = "prometheus.auth.password";
+  public static final String REGION = "prometheus.auth.region";
+  public static final String ACCESS_KEY = "prometheus.auth.access_key";
+  public static final String SECRET_KEY = "prometheus.auth.secret_key";
+
+
+  @Override
+  public ConnectorType getConnectorType() {
+    return ConnectorType.PROMETHEUS;
+  }
+
+  @Override
+  public StorageEngine getStorageEngine(String catalogName, Map<String, String> requiredConfig)
+      throws URISyntaxException {
+    validateFieldsInConfig(requiredConfig, Set.of(URI));
+    PrometheusClient prometheusClient =
+        new PrometheusClientImpl(getHttpClient(requiredConfig),
+            new URI(requiredConfig.get(URI)));
+    return new PrometheusStorageEngine(prometheusClient);
```

Comment:
Probably I asked this before but just want to confirm:

1. Because each storage engine is singleton and maintained in `CatalogServiceImpl`, is `PrometheusClient` thread-safe to be shared across queries in parallel?
2. Do we need to close the client connection properly (ex. when cluster shutdown)?

Reply:
1. Yes, thread safe for queries in parallel.https://stackoverflow.com/questions/48532860/is-it-thread-safe-to-make-calls-to-okhttpclient-in-parallel
2. will verify that.

Reply:
https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/ 
Shutdown isn't necessary, will be released when the threads become idle.